homework/index.js

```diff
@@ -29,23 +24,110 @@
     return elem;
   }
 
+  function createDetailLine(ul, labelText, value) {
+    const li = createAndAppend('li', ul, { class: 'list' });
+    createAndAppend('h3', li, { text: `${labelText}` });
+    createAndAppend('span', li, { text: value });
+    return li;
+  }
+
+  function formatDate(date) {
+    return new Date(date).toLocaleString();
+  }
+
   function renderRepoDetails(repo, ul) {
-    createAndAppend('li', ul, { text: repo.name });
+    ul.innerHTML = ' ';
+    const first = createDetailLine(ul, 'Repository', '');
+    createAndAppend('a', first.lastChild, {
+      text: repo.name,
+      href: repo.html_url,
+      target: '_blank',
+    });
+    createDetailLine(ul, 'Description', repo.description);
+    createDetailLine(ul, 'Forks', repo.forks);
+    createDetailLine(ul, 'Updated', formatDate(repo.updated_at));
   }
 
-  function main(url) {
-    fetchJSON(url, (err, repos) => {
-      const root = document.getElementById('root');
-      if (err) {
-        createAndAppend('div', root, {
-          text: err.message,
-          class: 'alert-error',
+  async function renderContributors(repositories, ul) {
+    const url = repositories.contributors_url;
+    try {
+      const data = await fetchJSON(url);
```

Comment:
The word `data` is rather meaningless. The usage of that word should be reserved for cases where you absolutely know nothing about the nature of the data it refers to. In this case however, you know that you are fetching an array of contributors. Therefore, you should use the more precise name `contributors` here.

Then, your `.forEach()` down below can read as:

```js
 contributors.forEach(contributor => {
```

Reply:
thank you for feedbacks , i will review and update those problems.